webots_ros2_driver/src/Driver.cpp

```diff
@@ -38,6 +40,24 @@ int main(int argc, char **argv) {
 
   std::shared_ptr<webots_ros2_driver::WebotsNode> node = std::make_shared<webots_ros2_driver::WebotsNode>(robotName);
   node->init();
+
+  // The parent process must be ros2 run. Declaring from launch file is deprecated. Remove with 2024.0.0.
+  const int BUFFER_SIZE = 4096;
+  const pid_t parentPid = getppid();
+  char processFile[BUFFER_SIZE];
+  sprintf(processFile, "/proc/%d/cmdline", parentPid);
+  int fd = open(processFile, O_RDONLY);
+  char *processName = (char *)malloc(BUFFER_SIZE);
```

Comment:
Memory leak: this is never free.
Why do we need to perform a malloc here? Can't we simply reuse the existing buffer (and give it a more generic name)?